### DISCUSSION TITLE
pyenv/pyenv-virtualenv messes up prompt?

### DISCUSSION THREAD

**Discussion by craighurley** [starship/starship#2955] [Q&A] [answered]
Can anyone help me understand why pyenv/pyenv-virtualenv messes up my starship prompt?

### Example

Starting zsh in `~/tmp/` is fine:
```
tmp on  master via 🐍 v3.9.2 (aws1) on ☁️  (eu-west-1)
❯ cat .python-version
aws1

tmp on  master via 🐍 v3.9.2 (aws1) on ☁️  (eu-west-1)
❯ pyenv versions
  system
  3.9.1
  3.9.2
  3.9.2/envs/aws1
  3.9.2/envs/default
* aws1 (set by /Users/hurley/tmp/.python-version)
  default
```

Once I `cd` out of `~/tmp/`, pyenv-virtualenv auto switches to the default virtualenv (expected), but now my prompt gets reset to something ugly (unexpected)
```
tmp on  master via 🐍 v3.9.2 (aws1) on ☁️  (eu-west-1)
❯ cd ~
(default) \h:\W \u$ pyenv versions
  system
  3.9.1
  3.9.2
  3.9.2/envs/aws1
  3.9.2/envs/default
  aws1
* default (set by /Users/hurley/.pyenv/version)
(default) \h:\W \u$ cd ~/tmp/
(aws1) \h:\W \u$
```

### Config

`~/.zshrc`:
```
eval "$(pyenv init --path)"
eval "$(pyenv init -)"
eval "$(pyenv virtualenv-init -)"
eval "$(starship init zsh)"
```

### Version info

macos: 11.4

zsh: 5.8 (x86_64-apple-darwin20.0)

pyenv: 2.0.4

pyenv-virtualenv: 1.1.5 (python -m venv)

starship:
```
starship 0.56.0
branch:
commit_hash:
build_time:2021-07-13 23:23:51
build_env:rustc 1.53.0,
```

**Comment by andytom** [comment]
I am not sure but it looks like pyenv-virtualenv has the ability to set the prompt from digging about a bit you can disable that by setting the env var `PYENV_VIRTUALENV_DISABLE_PROMPT` to `1`.

**Reply by craighurley** [reply]
I tried that, but still have the same issue :(

`~/.zshrc`:

```
export PYENV_VIRTUALENV_DISABLE_PROMPT=1
eval "$(pyenv init --path)"
eval "$(pyenv init -)"
eval "$(pyenv virtualenv-init -)"
eval "$(starship init zsh)"
```

**Comment by craighurley** [comment, marked as answer]
I found the reason this issue happens, but I don't know why it behaves this way.

For now, I have been starting zsh from bash (my default shell).  In my `.bash_profile`, I commented out `eval "$(pyenv virtualenv-init -)"`.  After that change, when I start zsh, the zsh prompt behave as expected.

If I change my default shell to zsh, the prompt behaves as expected without having to edit my `.bash_profile`.
